# Capítulo 4 — Métricas de riesgo/retorno

**Objetivo:** CAGR, volatilidad, Sharpe, Sortino, Max DD, Calmar.

$$\mathrm{CAGR}=(V_T/V_0)^{1/T}-1,\quad \sigma=\mathrm{std}(R_m)\sqrt{12}$$
$$\mathrm{Sharpe}\approx(\mathrm{CAGR}-r_f)/\sigma$$


In [1]:
import sys
from pathlib import Path
import pandas as pd
ROOT = Path('..').resolve()
sys.path.insert(0, str(ROOT/'src'))
import portfolio_utils as pu
daily = pd.read_csv(ROOT/'data'/'precios_ajustados_diarios.csv', index_col=0, parse_dates=True)
monthly = pu.to_month_end(daily).loc[:'2026-09-30']
bt = pu.backtest_portfolio(monthly, pu.CURRENT_WEIGHTS, start='2016-01-31', end='2026-09-30', rebalance='A')
spy = pu.backtest_portfolio(monthly, {'SPY':1}, start='2016-01-31', end='2026-09-30', rebalance='N')
rf = pu.load_risk_free_monthly()
m_p = pu.performance_metrics(bt['value'], bt['return'], rf)
m_s = pu.performance_metrics(spy['value'], spy['return'], rf)
rows=['CAGR','Volatilidad','Max DD','Sharpe','Sortino','Valor final']
keys=['cagr','vol','max_dd','sharpe','sortino','end_value']
for r,k in zip(rows,keys):
    vp, vs = m_p[k], m_s[k]
    if k in ('cagr','vol','max_dd'):
        print(f'{r:20s} {100*vp:10.2f}% {100*vs:10.2f}%')
    elif k=='end_value':
        print(f'{r:20s} {vp:10.2f} {vs:10.2f}')
    else:
        print(f'{r:20s} {vp:10.2f} {vs:10.2f}')


Métrica              Portafolio        SPY
CAGR                     19.21%     14.89%
Volatilidad              17.85%     15.12%
Max DD                  -27.47%    -23.93%
Sharpe                     1.02       0.85
Sortino                    1.55       1.28
Valor final             66141.36   44481.57


**Interpretación:** mayor CAGR y Sharpe vs SPY a costa de Max DD más profundo (~−27.5% vs ~−24%).

## Conclusiones
- Validación PDF casi exacta; periodo a sep-2026 refuerza el sesgo growth/semis.

## Ejercicios
1. Sharpe con rf=0.
2. Sortino vs Sharpe en 2022.
3. Calmar = CAGR/|Max DD|.
